connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaConfigBackingStoreTest.java

```diff
@@ -842,6 +842,7 @@ public void testBackgroundConnectorDeletion() throws Exception {
         // Task configs for the deleted connector should also be removed from the snapshot
         assertEquals(Collections.emptyList(), configState.allTaskConfigs(CONNECTOR_IDS.get(0)));
         assertEquals(0, configState.taskCount(CONNECTOR_IDS.get(0)));
+        assertEquals(0, configStorage.deferredTaskUpdates.size());
```

Comment:
Nit: it's better if we can do a comparison of the entire map here, since that provides better error messages if the assertion fails.

Also, we should add a comment explaining why we have this check since it may get removed in a future change if it's unclear why this is necessary.

```suggestion
        // Make sure the deleted connector is removed from the map in order to prevent unbounded growth
        assertEquals(Collections.emptyMap(), configStorage.deferredTaskUpdates);
```